File: specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabric/preview/2025-03-01-preview/examples-js/RedeployNodes_UD_example.js
const {
  ServiceFabricManagedClustersManagementClient,
} = require("@azure/arm-servicefabricmanagedclusters");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to redeploys one or more nodes on the node type. It will disable the fabric nodes, trigger a shut down on the VMs, move them to a new node, and power them back on.
 *
 * @summary redeploys one or more nodes on the node type. It will disable the fabric nodes, trigger a shut down on the VMs, move them to a new node, and power them back on.
 * x-ms-original-file: 2025-03-01-preview/RedeployNodes_UD_example.json
 */
async function redeployAllNodesByUpgradeDomain() {
  const credential = new DefaultAzureCredential();
  const subscriptionId = "00000000-0000-0000-0000-000000000000";
  const client = new ServiceFabricManagedClustersManagementClient(credential, subscriptionId);
  await client.nodeTypes.redeploy("resRg", "myCluster", "BE", {
    updateType: "ByUpgradeDomain",
  });
}

File: specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabric/ServiceFabricManagedClusters/preview/2025-10-01-preview/examples-java/RedeployNodes_UD_example.java

import com.azure.resourcemanager.servicefabricmanagedclusters.models.NodeTypeActionParameters;
import com.azure.resourcemanager.servicefabricmanagedclusters.models.UpdateType;

/**
 * Samples for NodeTypes Redeploy.
 */
public final class Main {
    /*
     * x-ms-original-file: 2025-10-01-preview/RedeployNodes_UD_example.json
     */
    /**
     * Sample code: Redeploy all nodes by upgrade domain.
     * 
     * @param manager Entry point to ServiceFabricManagedClustersManager.
     */
    public static void redeployAllNodesByUpgradeDomain(
        com.azure.resourcemanager.servicefabricmanagedclusters.ServiceFabricManagedClustersManager manager) {
        manager.nodeTypes().redeploy("resRg", "myCluster", "BE",
            new NodeTypeActionParameters().withUpdateType(UpdateType.BY_UPGRADE_DOMAIN),
            com.azure.core.util.Context.NONE);
    }
}
